specification/communication/data-plane/CallAutomation/stable/2025-06-15/communicationservicescallautomation.json

```diff
@@ -0,0 +1,4456 @@
+        }
+      }
+    },
+    "MuteParticipantsResult": {
+      "description": "The result payload for muting participants from the call.",
+      "type": "object",
+      "properties": {
+        "operationContext": {
+          "description": "The operation context provided by client.",
+          "type": "string"
+        }
+      }
+    },
+    "CancelAddParticipantRequest": {
+      "description": "Request payload for cancelling add participant request.",
+      "required": [
+        "invitationId"
+      ],
+      "type": "object",
+      "properties": {
+        "invitationId": {
+          "description": "Invitation ID used to add a participant.",
+          "type": "string"
+        },
+        "operationContext": {
+          "description": "Used by customers when calling mid-call actions to correlate the request to the response event.",
+          "type": "string"
+        },
+        "operationCallbackUri": {
+          "description": "Set a callback URI that overrides the default callback URI set by CreateCall/AnswerCall for this operation.\r\nThis setup is per-action. If this is not set, the default callback URI set by CreateCall/AnswerCall will be used.",
+          "type": "string"
+        }
+      }
+    },
+    "CancelAddParticipantResponse": {
+      "description": "Response payload for cancel add participant request.",
+      "type": "object",
+      "properties": {
+        "invitationId": {
+          "description": "Invitation ID used to cancel the add participant action.",
+          "type": "string"
+        },
+        "operationContext": {
+          "description": "The operation context provided by client.",
+          "type": "string"
+        }
+      }
+    },
+    "RecordingContentType": {
+      "description": "The content type of call recording.",
+      "enum": [
+        "audio",
+        "audioVideo"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "RecordingContentType",
+        "modelAsString": true
+      }
+    },
+    "RecordingChannelType": {
+      "description": "The channel type of call recording.",
+      "enum": [
+        "mixed",
+        "unmixed"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "RecordingChannelType",
+        "modelAsString": true
+      }
+    },
+    "RecordingFormatType": {
+      "description": "The format type of call recording.",
+      "enum": [
+        "wav",
+        "mp3",
+        "mp4"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "RecordingFormatType",
+        "modelAsString": true
+      }
+    },
+    "StartCallRecordingRequest": {
+      "description": "The request payload start for call recording operation with call locator.",
+      "type": "object",
+      "properties": {
+        "callLocator": {
+          "$ref": "#/definitions/CallLocator",
+          "description": "The call locator. (Only one of callLocator or callConnectionId to be used)"
+        },
```

Comment:
In 2025-05-01, this property is "required"
```
      "required": [
        "callLocator"
      ],
```

So this is also a break. Though you probably intend this for adding `callConnectionId`.

Reply:
Thank you for pointing this out. The change from `required` to `optional` for `callLocator` was intentional and was previously approved. As you mentioned, the API still supports usage of `callLocator`, but now also gives clients the option to use `callConnectionId` instead. This makes the API more flexible while maintaining backward compatibility.

Please let me know if you have further concerns or need additional clarification.

cc: @khannavikas